boards/arm/wio_terminal/wio_terminal.dts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2021 Szymon Duchniewicz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <atmel/samd5xx19.dtsi>
+
+/ {
+	model = "Wio Terminal";
+	compatible = "seeed,wio-terminal";
+
+	chosen {
+	zephyr,console = &sercom2;
+	zephyr,shell-uart = &sercom2;
+	zephyr,sram = &sram0;
+	zephyr,flash = &flash0;
+	zephyr,code-partition = &code_partition;
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+	led0 = &led0;
+	sw0 = &user_button_0;
+	sw1 = &user_button_1;
+	sw2 = &user_button_2;
+	};
+
+	leds {
+	compatible = "gpio-leds";
+	led0: led_0 {
+	gpios = <&porta 15 0>;
+	label = "LED";
+	};
+	gpio_keys {
+	compatible = "gpio-keys";
+	user_button_0: button_0 {
+	label = "User Button 0";
+	gpios = <&portc 26 1>;
+	};
+	user_button_1: button_1 {
+	label = "User Button 1";
+	gpios = <&portc 27 1>;
+	};
+	user_button_2: button_2 {
+	label = "User Button 2";
+	gpios = <&portc 28 1>;
+	};
+	};
+	};
+};
+
+&cpu0 {
+	clock-frequency = <120000000>;
+};
+
+&sercom2 {
+	status = "okay";
+	compatible = "atmel,sam0-uart";
+	current-speed = <115200>;
+	rxpo = <1>;
+	txpo = <0>;
+	pinctrl-0 = <&pb27c_sercom2_pad1 &pb26c_sercom2_pad0>;
+};
+
+
+
+&dmac {
+	status = "okay";
+};
```

Comment:
Try leaving out this snippet from the DTS. The ASYNC UART API for the SAM0 is conditional on having an enabled DMA controller (see https://github.com/zephyrproject-rtos/zephyr/blob/aa825d77b08a4b4c1ca660dc7449517701c00227/drivers/serial/Kconfig.sam0#L15). If I enabled this on the `adafruit_feather_m0_basic_proto`, the tests fail for me as well.

Reply:
It still fails even without enabling 'dmac'. How do you build it for `adafruit_feather_m0_basic_proto`? I run `west build -p always -b adafruit_feather_m0_basic_proto zephyr/tests/drivers/uart/uart_async_api` and it fails the build.

Reply:
Try `./scripts/twister -p adafruit_feather_m0_basic_proto`. With the above command, you force that test to be run, even though it should be filtered out for this board in `testcase.yaml`.

Reply:
Ok, so this test should be filtered out for` adafruit_feather_m0_basic_proto`? In this case should I also filter it out for the wio terminal? I saw that there are some board specifc overlays, like this one for `atsamd21_xpro` https://github.com/zephyrproject-rtos/zephyr/blob/aa825d77b08a4b4c1ca660dc7449517701c00227/tests/drivers/uart/uart_async_api/boards/atsamd21_xpro.overlay#L4 where the DMA channels are redefined for async operations and tx-rx are set to the same pin.

Reply:
Cause indeed these tests got skipped when running them with twister. (for `adafruit_feather_m0_basic_proto`)

Reply:
@henrikbrixandersen I also ran `./scripts/twister -p adafruit_itsybitsy_m4_express -T tests/drivers/uart/` to check if it also skips the tests for that board, but they don't get skipped and it just fails the `drivers.uart` tests,

Reply:
if SAM0 DMAC is enabled you need configure all DMAC channels.
There are examples at https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/uart/uart_async_api/boards